### ISSUE TITLE
`--wait-for-jobs` causes werf not to track resources during `helm` `install`/`upgrade`

### ISSUE THREAD

**Issue by felipecrs** [werf/werf#5961]
### Before proceeding

- [X] I didn't find a similar [issue](https://github.com/werf/werf/issues)

### Version

v1.2.288

### How to reproduce

1. Confirm resource tracking works with:

```console
$ ./werf helm upgrade --install --wait jenkins https://github.com/jenkinsci/helm-charts/releases/download/jenkins-5.0.12/jenkins-5.0.12.tgz --set persistence.enabled=false

┌ Waiting for resources to become ready
│ ┌ Status progress
│ │ STATEFULSET                                                                                 REPLICAS         READY              UP-TO-DATE
│ │ jenkins                                                                                     1/1              1                  1
│ │ RESOURCE                                                            NAMESPACE             WATCHING FOR FIELD
│ │ ServiceAccount/jenkins                                              default               -
│ │ Secret/jenkins                                                      default               -
│ │ ConfigMap/jenkins                                                   default               -
│ │ ConfigMap/jenkins-jenkins-jcasc-config                              default               -
│ │ Role/jenkins-schedule-agents                                        default               -
│ │ Role/jenkins-casc-reload                                            default               -
│ │ RoleBinding/jenkins-schedule-agents                                 default               -
│ │ RoleBinding/jenkins-watch-configmaps                                default               -
│ │ Service/jenkins-agent                                               default               -
│ │ Service/jenkins                                                     default               -
│ └ Status progress
└ Waiting for resources to become ready (3.00 seconds)
Release "jenkins" has been upgraded. Happy Helming!
NAME: jenkins
LAST DEPLOYED: Wed Feb  7 18:15:15 2024
NAMESPACE: default
STATUS: deployed
REVISION: 2
NOTES:
1. Get your 'admin' user password by running:
  kubectl exec --namespace default -it svc/jenkins -c jenkins -- /bin/cat /run/secrets/additional/chart-admin-password && echo
2. Get the Jenkins URL to visit by running these commands in the same shell:
  echo http://127.0.0.1:8080
  kubectl --namespace default port-forward svc/jenkins 8080:8080

3. Login with the password from step 1 and the username: admin
4. Configure security realm and authorization strategy
5. Use Jenkins Configuration as Code by specifying configScripts in your values.yaml file, see documentation: http://127.0.0.1:8080/configuration-as-code and examples: https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos

For more information on running Jenkins on Kubernetes, visit:
https://cloud.google.com/solutions/jenkins-on-container-engine

For more information about Jenkins Configuration as Code, visit:
https://jenkins.io/projects/jcasc/


NOTE: Consider using a custom image with pre-installed plugins
#################################################################################
######   WARNING: Persistence is disabled!!! You will lose your data when   #####
######            the Jenkins pod is terminated.                            #####
#################################################################################
```

2. Note how it does not work if you add `--wait-for-jobs`:

```console
$ ./werf helm upgrade --install --wait --wait-for-jobs --set persistence.enabled=false jenkins https://github.com/jenkinsci/helm-charts/releases/download/jenkins-5.0.12/jenkins-5.0.12.tgz
Release "jenkins" has been upgraded. Happy Helming!
NAME: jenkins
LAST DEPLOYED: Wed Feb  7 18:15:32 2024
NAMESPACE: default
STATUS: deployed
REVISION: 3
NOTES:
1. Get your 'admin' user password by running:
  kubectl exec --namespace default -it svc/jenkins -c jenkins -- /bin/cat /run/secrets/additional/chart-admin-password && echo
2. Get the Jenkins URL to visit by running these commands in the same shell:
  echo http://127.0.0.1:8080
  kubectl --namespace default port-forward svc/jenkins 8080:8080

3. Login with the password from step 1 and the username: admin
4. Configure security realm and authorization strategy
5. Use Jenkins Configuration as Code by specifying configScripts in your values.yaml file, see documentation: http://127.0.0.1:8080/configuration-as-code and examples: https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos

For more information on running Jenkins on Kubernetes, visit:
https://cloud.google.com/solutions/jenkins-on-container-engine

For more information about Jenkins Configuration as Code, visit:
https://jenkins.io/projects/jcasc/


NOTE: Consider using a custom image with pre-installed plugins
#################################################################################
######   WARNING: Persistence is disabled!!! You will lose your data when   #####
######            the Jenkins pod is terminated.                            #####
#################################################################################
```

### Result

See above.

### Expected result

It should work the same way it works without it.

**Comment by ilya-lesikov** [comment]
Should be fixed in 1.2.289

**Comment by felipecrs** [comment]
That's awesome. Thanks a lot!
